### ISSUE TITLE
_fields within deep Query not working

### ISSUE THREAD

**Issue by u12206050** [directus/directus#22800]
### Describe the Bug

Using the `_fields` on a relationship within the `deep` property on Query doesn't seem to do anything. The specified fields are not selected.

### To Reproduce

```
const eventSrv = new ItemsService('events');
const events = await eventSrv.readByQuery({
	fields: ['id', 'title', 'sessions'], // Have tried without sessions
	filter: {
		sessions: {
			status: { _eq: EVENT_STATUS.PUBLISHED },
		},
	},
	limit: 1,
	deep: {
		sessions: {
			_fields: ['id', 'start', 'end'],
		},
	},
});
```

console.log(events[0].sessions); // Expected to contain array of sessions with the keys `id, start, end`, instead it only contains the session id

### Directus Version

10.10.7

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by hanneskuettner** [comment]
Looking at the code I can tell you that the `_fields` within a deep query are indeed not considered. (See https://github.com/directus/directus/blob/fc8801f0c085a08af1eaff8d20ea293bc85ec720/api/src/utils/get-ast-from-query.ts#L263). So I would definitely agree that this behavior should be documented, as the current docs are a bit misleading.

But at the same time I'm wondering why you need to set the fields in the `deep` query parameter, when this would be equivalent:

`fields: ['id', 'title', 'sessions.id', 'sessions.start', 'sessions.end']`

I can see that is nicer to write, as opposed to having to prefix all fields with `sessions.`, but would you expect any function differences from your query?

**Comment by u12206050** [comment]
The nice thing with using the deep fields is that we can type it. eg. 
```
{
  fields: ['id', 'name'] as keyof Person,
  deep: {
     relations: {
       _fields: ['id', 'related_id', 'relationship'] as keyof Relationship
     } 
  }
}
```

**Comment by rijkvanzanten** [comment]
That was initially designed that way to avoid conflicts / confusion between having `relations.id` and `relations deep _fields ['name']`, but could indeed be nice for typing reasons

**Comment by u12206050** [comment]
On the note of typing, I spent the whole day to finally solve being able to get type completion for the entire Query object, including the fields property with dot delimited strings. Feast your eyes ❤️ 

![QueryTypes](https://github.com/directus/directus/assets/6641242/e2a199fe-d4e5-443e-b7a9-74dbed93aae2)

**Comment by datner** [comment]
Isn't it supposed to be like this?
```ts
const eventSrv = new ItemsService('events');
const events = await eventSrv.readByQuery({
	fields: ['id', 'title', { sessions: ['id', 'start', 'end']}],
	filter: {
		sessions: {
			status: { _eq: EVENT_STATUS.PUBLISHED },
		},
	},
	limit: 1,
});
```
it's typed and everything

**Comment by u12206050** [comment]
> Isn't it supposed to be like this?
> 
> ```ts
> 
> const eventSrv = new ItemsService('events');
> 
> const events = await eventSrv.readByQuery({
> 
> 	fields: ['id', 'title', { sessions: ['id', 'start', 'end']}],
> 
> 	filter: {
> 
> 		sessions: {
> 
> 			status: { _eq: EVENT_STATUS.PUBLISHED },
> 
> 		},
> 
> 	},
> 
> 	limit: 1,
> 
> });
> 
> ```
> 
> it's typed and everything

No, you are missing the 'deep' property on the query in which you would be able to add relational filters, aggregation and supposably fields.

**Comment by datner** [comment]
yes the filter is incorrect, I was actually just talking about the `fields`, I should have been clearer.
this should produce
```
id
title
sessions.id
sessions.start
sessions.end
```
as the fields, no?

**Comment by licitdev** [comment]
Closing the issue as `fields` is not supported within `deep`. It's documented in https://github.com/directus/docs/blob/9dfdca887aedd2b8b4519d61c1fcc7bfe86520d3/content/guides/4.connect/3.query-parameters.md?plain=1#L402
Thank you.
